src/lib.rs

```diff
@@ -344,6 +360,31 @@ pub trait LanguageServer: Send + Sync + 'static {
     ///
     /// [`textDocument/documentHighlight`]: https://microsoft.github.io/language-server-protocol/specifications/specification-3-15/#textDocument_documentHighlight
     fn document_highlight(&self, params: TextDocumentPositionParams) -> Self::HighlightFuture;
+
+    /// The [`textDocument/signatureHelp`] request is sent from the client to the server to request
+    /// signature information at a given cursor position.
+    ///
+    /// [`textDocument/signatureHelp`]: https://microsoft.github.io/language-server-protocol/specifications/specification-3-15/#textDocument_signatureHelp
+    fn signature_help(&self, params: TextDocumentPositionParams) -> Self::SignatureHelpFuture;
+
+    /// The [`textDocument/implementation`] request is sent from the client to the server to resolve
+    /// the implementation location of a symbol at a given text document position.
+    ///
+    /// The result type [`GotoImplementationResponse::Link`] got introduced with version 3.14.0 and
+    /// requires client-side support. It can be returned if the client set the following
+    /// field to `true` in the [`initialize`] method:
+    ///
+    /// ```text
+    /// InitializeParams::capabilities::text_document::implementation::link_support
+    /// ```
+    ///
+    /// [`textDocument/implementation`]: https://microsoft.github.io/language-server-protocol/specifications/specification-3-15/#textDocument_implementation
+    /// [`GotoImplementationResponse::Link`]: https://docs.rs/lsp-types/0.63.1/lsp_types/request/enum.GotoDefinitionResponse.html
+    /// [`initialize`]: #tymethod.initialize
+    fn goto_implementation(
```

Comment:
This method should be moved to immediately below `LanguageServer::goto_type_definition()`.